File: FSUIKit/Classes/FSTabView.m
//
//  FSTabView.m
//  FSUIKit
//
//  Created by FudonFuchina on 2021/8/7.
//

#import "FSTabView.h"

@implementation FSTabView

- (void)layoutSubviews {
    [super layoutSubviews];
    _label.frame = self.bounds;
}

- (UILabel *)label {
    if (!_label) {
        _label = [[UILabel alloc] initWithFrame:self.bounds];
        _label.textAlignment = NSTextAlignmentCenter;
        [self addSubview:_label];
    }
    return _label;
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: FSUIKit/Classes/Swift/FSTabView.swift
// FSTabViewS.swift
// Translated from FSTabView.h/m

import UIKit

public class FSTabView: FSView {
    
    private var _label: UILabel?
    
    public var label: UILabel {
        if _label == nil {
            let lbl = UILabel(frame: bounds)
            lbl.textAlignment = .center
            addSubview(lbl)
            _label = lbl
        }
        return _label!
    }
    
    public var selected: Bool = false {
        didSet {
            selectedState?(self, selected)
        }
    }
    
    public var selectedState: ((FSTabView, Bool) -> Void)?
    
    public override func layoutSubviews() {
        super.layoutSubviews()
        _label?.frame = bounds
    }
}
